frontend/src/pages/modelServing/customServingRuntimes/CustomServingruntimeAddTemplate.tsx

```diff
@@ -0,0 +1,114 @@
+import * as React from 'react';
+import {
+  ActionList,
+  ActionListItem,
+  Alert,
+  AlertActionCloseButton,
+  Breadcrumb,
+  BreadcrumbItem,
+  Button,
+  Stack,
+  StackItem,
+} from '@patternfly/react-core';
+import { CodeEditor, Language } from '@patternfly/react-code-editor';
+import { Link, useNavigate } from 'react-router-dom';
+import ApplicationsPage from '~/pages/ApplicationsPage';
+import { TemplateKind } from '~/k8sTypes';
+import { createServingRuntimeTemplate } from '~/api';
+import { useDashboardNamespace } from '~/redux/selectors';
+
+type CustomServingRuntimesAddTemplateProps = {
+  existingCustomServingRuntime?: TemplateKind;
+};
+
+const CustomServingRuntimesAddTemplate: React.FC<CustomServingRuntimesAddTemplateProps> = ({
+  existingCustomServingRuntime,
+}) => {
+  const { dashboardNamespace } = useDashboardNamespace();
+  const [code, setCode] = React.useState('');
+  const [loading, setIsLoading] = React.useState(false);
+  const [error, setError] = React.useState<Error | undefined>(undefined);
+  const navigate = useNavigate();
+
+  return (
+    <ApplicationsPage
+      title="Add serving runtime"
+      description="Add a new runtime that will be available for Data Science users on this cluster"
+      breadcrumb={
+        <Breadcrumb>
+          <BreadcrumbItem render={() => <Link to="/servingRuntimes">Serving Runtimes</Link>} />
+          <BreadcrumbItem isActive>Add serving runtime</BreadcrumbItem>
+        </Breadcrumb>
+      }
+      loaded
+      empty={false}
+      provideChildrenPadding
+    >
+      <Stack hasGutter>
+        <StackItem>
+          <CodeEditor
```

Comment:
You could set `emptyStateTitle`, `emptyStateBody`, and `emptyStateButton` to the texts shown in the mockup. For example:
```
emptyStateTitle="Add a serving runtime"
emptyStateBody="Drag a file here, upload files, or start from scratch."
emptyStateButton="Upload files"
```

Reply:
You may also want to set
```
options={{ tabSize: 2 }}
```
Now it's 4 when you hit `tab` key, but it should be 2 as I tested in the OpenShift console.